dynamic_programming/max_non_adjacent_sum.py

```diff
@@ -0,0 +1,52 @@
+from typing import List
+
+
+def max_non_adj_sum(nums: List[int]) -> int:
+    '''
+    Function to find the maximum non-adjacent sum of the elements in the input list
+
+    Parameters
+    ----------
+    nums : List[int]
+        List of integers for the maximum non-adjacent sum computation
+
+    Returns
+    -------
+    int
+        maximum non-adjacent sum
```

Comment:
```suggestion
    Find the maximum non-adjacent sum of the integers in the nums input list
```
Type hints and careful naming of functions and variables makes a lot of comments unnecessary boilerplate.
1. `def` tells us that this is a function.
2. The self-documenting function name tells us what this function does.
3. The type hints already tell us that `nums` is a list of integers.
4. The type hint on the return value tells us that we will return an it.

Reply:
Looking into it